sops/__init__.py

```diff
@@ -590,17 +595,25 @@ def encrypt_key_with_kms(key, entry):
         print("Unable to find region from ARN '%s' in entry" %
               entry['arn'], file=sys.stderr)
         return entry
-    kms = boto3.client('kms', region_name=region)
+    # if there are no role to assume, return the client directly
+    if not ('role' in entry):
+        return boto3.client('kms', region_name=region)
+    # otherwise, create a client using temporary tokens that assume the role
+    client = boto3.client('sts')
+    role = client.assume_role(RoleArn=entry['role'],
+                              RoleSessionName='sops@'+gethostname())
```

Comment:
I'd prefer to see 603 and 604 in a try block with an appropriate error message, because i feel like this one will happen fairly often.